### PR TITLE
added cross product for 2D and 4D following discussion at 

### PR DESCRIPTION
**Pull request by micampbell**
https://github.com/dotnet/corefx/issues/35434

### CODE REVIEW

File: src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs [hunk 1]
```diff
@@ -50,7 +50,7 @@ public Vector2(float x, float y)
         /// <param name="array">The destination array.</param>
         [Intrinsic]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public readonly void CopyTo(float[] array)
+        public void CopyTo(float[] array)
```
Comment (gfoidl) [inline src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs]: Why did you remove the `readonly`?

Comment (micampbell) [reply src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs]: ah crud. That was a mistake. I should have been more careful. The only change was the addition of the function (one to each file).

Comment (micampbell) [reply src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs]: I must've been working from an old version

Comment (tannergooding) [reply src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs]: Could you fixup the diff? Once that is done, I think it looks basically complete modulo tests :)

**Comment by maryamariyan** [comment]
@micampbell thank you for this PR.
Looks like the pull request is almost ready (some comments remaining to address). 
Are you planning to update this soon?

File: src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs [hunk 2]
```diff
@@ -9,7 +9,7 @@ namespace System.Numerics
     // This file contains the definitions for all of the JIT intrinsic methods and properties that are recognized by the current x64 JIT compiler.
     // The implementation defined here is used in any circumstance where the JIT fails to recognize these members as intrinsic.
     // The JIT recognizes these methods and properties by name and signature: if either is changed, the JIT will no longer recognize the member.
-    // Some methods declared here are not strictly intrinsic, but delegate to an intrinsic method. For example, only one overload of CopyTo()
```
Comment (micampbell) [inline src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs]: returning comment to newer one before diff

File: src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs [hunk 3]
```diff
@@ -50,7 +50,7 @@ public Vector2(float x, float y)
         /// <param name="array">The destination array.</param>
         [Intrinsic]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public readonly void CopyTo(float[] array)
```
Comment (micampbell) [inline src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs]: re-adding readonly

File: src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs [hunk 4]
```diff
@@ -64,7 +64,7 @@ public Vector2(float x, float y)
         /// <exception cref="ArgumentException">If number of elements in source vector is greater than those available in destination array
         /// or if there are not enough elements to copy.</exception>
         [Intrinsic]
-        public readonly void CopyTo(float[] array, int index)
```
Comment (micampbell) [inline src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs]: re-adding readonly

File: src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs [hunk 5]
```diff
@@ -89,7 +89,7 @@ public Vector2(float x, float y)
         /// <param name="other">The Vector2 to compare this instance to.</param>
         /// <returns>True if the other Vector2 is equal to this instance; False otherwise.</returns>
         [Intrinsic]
-        public readonly bool Equals(Vector2 other)
```
Comment (micampbell) [inline src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs]: re-adding readonly

File: src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs [hunk 6]
```diff
@@ -144,7 +160,7 @@ public static Vector2 Max(Vector2 value1, Vector2 value2)
         /// Returns a vector whose elements are the absolute values of each of the source vector's elements.
         /// </summary>
         /// <param name="value">The source vector.</param>
-        /// <returns>The absolute value vector.</returns>
```
Comment (micampbell) [inline src/System.Numerics.Vectors/src/System/Numerics/Vector2_Intrinsics.cs]: re-adding readonly

File: src/System.Numerics.Vectors/src/System/Numerics/Vector4_Intrinsics.cs [hunk 7]
```diff
@@ -9,7 +9,7 @@ namespace System.Numerics
     // This file contains the definitions for all of the JIT intrinsic methods and properties that are recognized by the current x64 JIT compiler.
     // The implementation defined here is used in any circumstance where the JIT fails to recognize these members as intrinsic.
     // The JIT recognizes these methods and properties by name and signature: if either is changed, the JIT will no longer recognize the member.
-    // Some methods declared here are not strictly intrinsic, but delegate to an intrinsic method. For example, only one overload of CopyTo()
```
Comment (micampbell) [inline src/System.Numerics.Vectors/src/System/Numerics/Vector4_Intrinsics.cs]: re-adding readonly

**Comment by micampbell** [comment]
Sorry for the confusing extra request (I'm new to this level of githusbandry). Hopefully, it is a clear addition of two small methods and no extra changes.
